src/components/Text/Text.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useEffect } from 'react';
+
+interface ITextTypes {
+  value: string;
+  placeholder: string;
+  error: boolean;
+  onChange: Function;
+}
+
+function Text({
+  value = '',
+  placeholder = 'Placeholder',
+  onChange,
+  error,
+  ...rest
+}): ITextTypes {
+  const [inputValue, setInputValue] = useState<string>(value);
+  const changeInputValue = (e) => {
+    setInputValue(e.target.value);
+    if (onChange) {
+      onChange(e);
+    }
+  };
+  useEffect(() => {
+    setInputValue(value);
+  }, [value]);
+  return (
+    <div className={`input-text${error ? '-error' : ''}`}>
```

Comment:
I'm getting a linter error here about the props not conforming `ITextType`.  I couldn't find any easy answers to this one, you'll need to research.  I assume it has something to do with returning a container `div` rather than a straight `input`.